### PR TITLE
Replace secondary image with carousel

### PR DESCRIPTION
**Pull request by Omicron7**

### CODE REVIEW

File: src/app/designationEditor/carouselModal/carouselModal.tpl.html [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+          <div class="btn-group-vertical btn-group-sm" role="group">
+            <button class="btn btn-default"
+                    type="button"
+                    aria-label="Remove image"
+                    ng-disabled="$index == 0"
+                    ng-click="$ctrl.reorderImageInCarousel($index, $index-1)"><i class="fa fa-sort-up"></i>
+            </button>
+            <button class="btn btn-default"
+                    type="button"
+                    aria-label="Remove image"
```
Comment (OzzieOrca) [inline src/app/designationEditor/carouselModal/carouselModal.tpl.html]: This label and the one above it are wrong.

File: src/app/designationEditor/designationEditor.component.js [hunk 2]
```diff
@@ -137,31 +144,78 @@ class DesignationEditorController {
...
+          return photoLocation;
+        },
+        selectedPhoto: () => {
+          return selectedUrls;
+        }
+      }
+    };
+    this.$uibModal.open(modalOptions).result.then((data) => {
+      const selectedUrls = [];
+      angular.forEach(data.selected, (selectedPhoto) => {
```
Comment (OzzieOrca) [inline src/app/designationEditor/designationEditor.component.js]:
```js
const selectedUrls = data.selected.map(({ url }) => url);
```

File: src/app/designationEditor/designationEditor.component.js [hunk 3]
```diff
@@ -243,24 +314,45 @@ class DesignationEditorController {
...
 
   isCampaign() {
     return this.designationContent && includes([
-        'Campaign'
-      ], this.designationContent.designationType);
+      'Campaign'
+    ], this.designationContent.designationType);
+  }
+
+  getImageUrls(obj, imageUrls) {
```
Comment (OzzieOrca) [inline src/app/designationEditor/designationEditor.component.js]: I'd prefer things to be more functional and immutable. I think it makes code much easier to follow. This is sorta shorthand, take whatever you want. I just think it's nicer to not have to pass the output array around and just have it return it.

```js
getImageUrls = obj =>
  Object.entries(obj).reduce(
    (acc, [key, value]) => [
      ...acc,
      ...(key === "fileReference"
        ? [value]
        : angular.isObject(obj)
        ? this.getImageUrls(value)
        : [])
    ],
    []
  );
```

File: src/app/designationEditor/designationEditor.component.js [hunk 4]
```diff
@@ -137,31 +144,78 @@ class DesignationEditorController {
...
         this.save();
-      }, angular.noop );
+      }, angular.noop);
+  }
+
+  selectPhotos(photoLocation, selectedPhotos) {
+    const imageUrls = [];
+    this.getImageUrls(selectedPhotos, imageUrls);
+    const selectedUrls = [];
+    for (let url in imageUrls) {
```
Comment (OzzieOrca) [inline src/app/designationEditor/designationEditor.component.js]:
```js
const selectedUrls = imageUrls.map(url => ({ url }));
```

File: src/app/designationEditor/designationEditor.component.js [hunk 5]
```diff
@@ -184,11 +181,7 @@ class DesignationEditorController {
       }
     };
     this.$uibModal.open(modalOptions).result.then((data) => {
-      const selectedUrls = [];
-      angular.forEach(data.selected, (selectedPhoto) => {
-        selectedUrls.push(selectedPhoto.url);
-      });
-      this.designationContent[photoLocation] = selectedUrls;
+      this.designationContent[photoLocation] = (data.selected || []).map(photo => photo.url);
```
Comment (OzzieOrca) [inline src/app/designationEditor/designationEditor.component.js]: 👍 `|| []`

File: src/app/designationEditor/designationEditor.component.js (hunk 3 above)
Comment (OzzieOrca) [reply src/app/designationEditor/designationEditor.component.js]: If you like what you have you could do this
```suggestion
  getImageUrls(obj, imageUrls = []) {
```
so you don't have to setup `imageUrls` beforehand. But I guess you'd still have to refactor to use return values.

**Comment by OzzieOrca** [comment]
Your branch has a bunch of already merged commits in it...

**Comment by Omicron7** [comment]
rebased this on master. I'm not sure what the timeline is for this feature, just keeping it up to date with other give development.

**Comment by wrandall22** [comment]
FDS reported some strange spacing issues with the designation editor:

Screen Shot 2020-02-20 at 5 14 11 PM

**Pull request merged**